### PR TITLE
refactor(callouts)!: convert to CSS modules

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1199185162145254/f)
🔍 [Preview Link](https://react-components-git-zscallouts-css-modules-hashicorp.vercel.app/components/callouts)

---

This PR updates `@hashicorp/react-callouts` to use CSS modules.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: pages/global.css [hunk 1]
```diff
@@ -3,7 +3,6 @@
 /* global component css imports */
 
 /* these should be moved to css modules */
-@import '../packages/callouts/styles/index.css';
```
Comment (zchsh) [inline pages/global.css:6]: 🔪 🎉

File: packages/callouts/partials/callout-item/index.test.js [hunk 2]
```diff
@@ -3,12 +3,6 @@ import propsDefault from './fixtures/default.json'
 import CalloutItem from './index.js'
 
 describe('<CalloutItem />', () => {
-  it('should render a `.callout-item` root element', () => {
```
Comment (zchsh) [inline packages/callouts/partials/callout-item/index.test.js:6]: Removed this test as it was mostly just testing that React works.

File: packages/callouts/style.module.css [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    while the flex "container" (.items) uses a negative margin to compensate for the extraneous padding at the edges of the grid
+    ---
+    Hopefully this helps explain how we arrive at the styles below, given the desired grid gaps from design side:
+    - .layout-two-up   : 32px gap                       =>  -16px margin & 16px padding
+    - .layout-three-up : 48px row gap, 64px column gap  =>  -24px -32px margin & 24px 32px padding
+    - .layout-four-up  : 48px row gap, 64px column gap  =>  -24px -32px margin & 24px 32px padding
+    ---
+    With this approach in place, we can also change the number of "grid columns" by modifying the % width of the "wrapper" items
+    The columns vary per layout and viewport size. 50% = 2 colums, 33.33% = 3 columns, 25% = 4 columns, etc
+    */
```
Comment (jescalan) [inline packages/callouts/style.module.css:90]: I wonder if we should tag this as tech debt now given that we no longer support ie11

**Review by jescalan — APPROVED** [review]

File: packages/callouts/style.module.css (hunk 3 above)
Comment (zchsh) [reply packages/callouts/style.module.css]: Great callout - added a task to [Use CSS grid in callouts](https://app.asana.com/0/1100423001970639/1200933695189575/f), and updated this comment with a backlink 👍

**Pull request merged**